eclipse-collections-code-generator/src/main/resources/impl/bag/immutable/immutablePrimitiveSingletonBag.stg

```diff
@@ -169,6 +171,14 @@ final class Immutable<name>SingletonBag implements Immutable<name>Bag, Serializa
             : <name>Bags.immutable.empty();
     }
 
+    @Override
+    public Immutable<name>Set selectUnique()
+    {
+        Immutable<name>Bag selected = this.selectByOccurrences(occurrences -> occurrences == 1);
+        Immutable<name>Set result = <name>Sets.immutable.ofAll(selected);
+        return result;
```

Comment:
This is a SingletonBag, so it will contain only 1 element and occurrences == 1
So, you can do this `return <name>Sets.immutable.with(this.value);`